subsys/instrumentation/CMakeLists.txt

```diff
@@ -0,0 +1,31 @@
+# Copyright 2023 Linaro
+# SPDX-License-Identifier: Apache-2.0
+
+zephyr_sources_ifdef(CONFIG_INSTRUMENTATION
+  common/instr_common.c
+  timestamp/timestamp.c
+  transport/uart.c
+)
+
+zephyr_sources_ifdef(CONFIG_INSTRUMENTATION_MODE_CALLGRAPH ringbuffer/ringbuffer.c)
+
+if (CONFIG_INSTRUMENTATION)
+  if(CMAKE_C_COMPILER_ID STREQUAL "GNU")
+    zephyr_sources(handlers/instr_handlers_gcc.c)
+  else()
+    message(FATAL_ERROR "CONFIG_INSTRUMENTATION unsupported by this compiler.")
+  endif()
+
+  if (CONFIG_INSTRUMENTATION_EXCLUDE_FUNCTION_LIST)
+	  zephyr_compile_options(-finstrument-functions-exclude-file-list=${CONFIG_INSTRUMENTATION_EXCLUDE_FUNCTION_LIST})
+  endif()
+
+  if (CONFIG_INSTRUMENTATION_EXCLUDE_FILE_LIST)
+	  zephyr_compile_options(-finstrument-functions-exclude-function-list=${CONFIG_INSTRUMENTATION_EXCLUDE_FILE_LIST})
```

Comment:
are these flags backwards?  `-functions-excluded-funciton-list=` be part of line 20 and `functions-exclude-file-list=` part of 24?  If they are not backwards, it may be worth adding a comment to explicitly call out why these are correctly located.